src/tests/testCacheManager.cc

```diff
@@ -20,6 +20,9 @@
  * test the CacheManager implementation
  */
 
+/// Define Chrooted variable to use Instance.o
+bool Chrooted = false;
```

Comment:
> I also added the global variable **Chrooted** to the tests to make them use Instance.o. Is it OK?

IMO that is fine as a workaround for globals. The "correct" code is a much larger change which should not be pulled into the scope of this PR.

However each addition should be commented differently. Like so:
```suggestion
// XXX: for Instance.cc linkage. keep in sync with main.cc
bool Chrooted = false;
```